src/com/company/EchoServer.java

```diff
@@ -0,0 +1,51 @@
+package com.company;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class EchoServer {
+
+    private ServerSocket serverSocket;
+    private int portNumber;
+    private ServerMessenger serverMessenger;
+
+    public EchoServer(ServerSocket serverSocket, int portNumber, ServerMessenger serverMessenger) {
```

Comment:
portNumber is an attribute of ServerSocket and therefore could be accessed via a getter in ServerSocket whenever it is needed in EchoServer e.g. to print it out

Reply:
done.